specs/exec-engine.md

```diff
@@ -46,6 +51,74 @@ To process deposited transactions safely, the deposits MUST be authenticated fir
 Deposited transactions MUST never be consumed from the transaction pool.
 *The transaction pool can be disabled in a deposits-only rollup*
 
+## Fees
+
+Sequenced transactions (i.e. not applicable to deposits) are charged with 3 types of fees:
+priority fees, base fees, and L1-cost fees.
+
+### Fee Vaults
+
+The three types of fees are collected in 3 distinct L2 fee-vault deployments for accounting purposes:
+fee payments are not registered as internal EVM calls, and thus distinguished better this way.
+
+These are hardcoded addresses, pointing at pre-deployed proxy contracts.
+The proxies are backed by vault contract deployments, based on `FeeVault`, to route vault funds to L1 securely.
+
+| Vault Name          | Predeploy                                             |
+|---------------------|-------------------------------------------------------|
+| Sequencer Fee Vault | [`SequencerFeeVault`](./predeploys#SequencerFeeVault) |
+| Base Fee Vault      | [`BaseFeeVault`](./predeploys#BaseFeeVault)           |
+| L1 Fee Vault        | [`L1FeeVault`](./predeploys#L1FeeVault)               |
+
+### Priority fees (Sequencer Fee Vault)
+
+Priority fees follow the [eip-1559] specification, and are collected by the fee-recipient of the L2 block.
+The block fee-recipient (a.k.a. coinbase address) is set to the Sequencer Fee Vault address.
```

Comment:
Perhaps note that the `COINBASE` opcode pushes this value to the stack. Side note: perhaps longer term we want to come up with a different solution such that priority fees can go to the block proposer so that MEV extraction is more similar to L1